static/js/util/profile_edit.js

```diff
@@ -220,6 +221,57 @@ export function boundCheckbox(keySet: string[], label: string|React$Element<*>):
   );
 }
 
+const onTelChange = R.curry((
+  keySet,
+  countryKeySet,
+  profile,
+  updateProfile,
+  validator,
+  newPhoneNumber,
+  countryObj
+) => {
+  let currentPhoneNumber = _.get(profile, keySet);
+  if ( currentPhoneNumber !== newPhoneNumber ) {
+    let clone = _.cloneDeep(profile);
+    _.set(clone, keySet, newPhoneNumber);
+    _.set(clone, countryKeySet, countryObj.iso2);
```

Comment:
Since the output formatting varies across countries I don't feel putting the burden of parsing that on the backend is the right choice since it's an implementation detail of the frontend, but I also don't want to make a mess working around this.

I think an easier path would be to parse the two parts out via regex? Reading the sources for the input I think `/\+([0-9]+)(.+)/` would probably do the trick (http://regexr.com/3f1a3) and is much less hacky than what you're describing. It should be easy to regenerate that string to initialize the input as well.